brunoise/streaming_save.py

```diff
@@ -124,41 +133,87 @@ def fill_dataset(self, frame):
         if self.i_in_plane == self.save_parameters.n_t:
             self.complete_plane()
 
+    def dump_metadata(self, file):
+        json.dump(
+            {
+                "shape_full": (
+                    self.save_parameters.n_t,
+                    self.i_block,
+                    *self.current_data.shape[2:],
+                ),
+                "shape_block": (
+                    self.save_parameters.n_t,
+                    1,
+                    *self.current_data.shape[2:],
+                ),
+                "crop_start": [0, 0, 0, 0],
+                "crop_end": [0, 0, 0, 0],
+                "padding": [0, 0, 0, 0],
+            },
+            file,
+        )
+
     def finalize_dataset(self):
-        with open(
+        if self.save_parameters.channel == "Both":
```

Comment:
Same here, am I overlooking something, or should there be a case for red and green individually as well?